### DISCUSSION TITLE
Error: Access token does not cover required scopes"

### DISCUSSION THREAD

**Discussion by streiman** [mastodon/mastodon#18318] [Q&A] [answered]
I installed a Mastodon instance using docker-compose behin Traefik. It seems to work, but when I first access the website mastodon.mydomain.com I receive a blank page only stating

{"error":"Error: Access token does not cover required scopes"}

If I reload, I get the Mastodon page. But every now and then when I click somewhere, I will get the blank browser window just with a {"error":"Error: Access token does not cover required scopes"} again.

I tried to change the Traefik config, but I am not really sure it is the cause for the trouble.

Anybody here able to point me to a way to solve that?

**Comment by agrimpelhuber** [comment, marked as answer]
Can you post your **Trafik** configuration (Labels) for the docker containers (web, streaming) from docker-compose?

Reason: The answer you see comes from the streaming container. When you configure **both containers** with

```
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.mastoston-streaming.entryPoints=websecure"
      - "traefik.http.routers.mastoston-streaming.rule=Host(`my.domain`)"
      [...]
```
Traefik randomly directs to either container. To fix this, the streaming container needs:
```
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.mastoston-streaming.entryPoints=websecure"
      - "traefik.http.routers.mastoston-streaming.rule=Host(`my.domain`) && PathPrefix(`/api/v1/streaming/`)"
      [...]
```
In this case the longer rule for the streaming API has priority. If it does not match, the remainder of requests hit the web container.

**Reply by streiman** [reply]
Thank you for your help and your explanation. I thought I did this already in my docker-compose:


   web:
    labels:
      - "traefik.enable=true"      
      - "traefik.http.routers.mastodon-secure.entrypoints=https"
      - "traefik.http.routers.mastodon-secure.rule=Host(`mastodon.mydomain.com`)"
      - "traefik.http.routers.mastodon-secure.tls.certresolver=http"
      - "traefik.http.services.mastodon.loadbalancer.server.port=3000"
      - "traefik.docker.network=proxy"
      - "traefik.http.middlewares.addprotoheader.headers.customresponseheaders.X-Forwarded-Proto=https"

   streaming:
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.mastodon-streaming-secure.entrypoints=https"
      - "traefik.http.routers.mastodon-streaming-secure.rule=(Host(`mastodon.mydomain.com`) && PathPrefix(`/api/v1/streaming`))"
      - "traefik.http.routers.mastodon-streaming-secure.tls.certresolver=http"
      - "traefik.http.services.mastodon.loadbalancer.server.port=4000"
      - "traefik.docker.network=proxy"

**Reply by streiman** [reply]
Oh, I think I found the error. I wrote

    - "traefik.http.services.mastodon.loadbalancer.server.port=4000"

Instead of 

    - "traefik.http.services.mastodon-streaming-secure.loadbalancer.server.port=4000"
    
  And now it seems to work.

**Reply by agrimpelhuber** [reply]
Hi,

different reason, same outcome - the answer you saw came from the wrong service, because the port mapping didn't work.

Great that it works now, and thanks for giving feedback on how you fixed it  - can you mark the question as answered?

**Reply by streiman** [reply]
I would love to mark the Q&A as answered, but - feeling a little stupid - can't find the checkbox to do so. :-/
